File: 275. H-Index II.cpp
class Solution {
public:
    int hIndex(vector<int>& citations) {
        int n = citations.size();
        int l = 0, r = n, m,x = 0;
        while (l<=r)
        {
            m = (l+r)/2;
            if (citations.end() - lower_bound(citations.begin(), citations.end(), m) >= m)
            {
                x = m;
                l = m + 1;
            }
            else
            {
                r = m - 1;
            }
        }
        return x;
    }
};

File: 275. H-Index II.py
class Solution:
    def hIndex(self, citations: List[int]) -> int:
        n = len(citations)
        l, r = 0, n
        while l < r:
            m = (l + r) >> 1
            if citations[m] < n - m:
                l = l + 1
            else:
                r = m
        return n - r